core/src/mindustry/logic/LogicOp.java

```diff
@@ -4,24 +4,32 @@
 import arc.util.*;
 
 public enum LogicOp{
+    //section arithmetic
     add("+", (a, b) -> a + b),
     sub("-", (a, b) -> a - b),
     mul("*", (a, b) -> a * b),
     div("/", (a, b) -> a / b),
     idiv("//", (a, b) -> Math.floor(a / b)),
     mod("%", (a, b) -> a % b),
+    pow("^", Math::pow),
+    //section boolean logic
     equal("==", (a, b) -> Math.abs(a - b) < 0.000001 ? 1 : 0, (a, b) -> Structs.eq(a, b) ? 1 : 0),
-    notEqual("not", (a, b) -> Math.abs(a - b) < 0.000001 ? 0 : 1, (a, b) -> !Structs.eq(a, b) ? 1 : 0),
+    notEqual("=/=", (a, b) -> Math.abs(a - b) < 0.000001 ? 0 : 1, (a, b) -> !Structs.eq(a, b) ? 1 : 0),
     lessThan("<", (a, b) -> a < b ? 1 : 0),
     lessThanEq("<=", (a, b) -> a <= b ? 1 : 0),
     greaterThan(">", (a, b) -> a > b ? 1 : 0),
     greaterThanEq(">=", (a, b) -> a >= b ? 1 : 0),
-    pow("^", Math::pow),
+    lnot("!", (a, b) -> a < 0.000001 ? 1 : 0, (a, b) -> a == null ? 1 : 0),
+    lor("||", (a, b) -> a < 0.000001 || b < 0.000001 ? 1 : 0, (a, b) -> a != null || b != null),
```

Comment:
Why is this needed?

Reply:
if (!cond || !ptr)
if (block || enemy)

Reply:
That's exactly what `or` already does.

Reply:
technically correct, but can be neither 0 nor 1, e.g. 0 | 2 = 2

Reply:
Why is that a problem? `0 | 2 = 2 != 0` -> will be evaluated as `true` in any condition